### PR TITLE
chore: `rs-dpp` and `wasm-dpp` updates for integration

### PR DESCRIPTION
**Pull request by jawid-h**
## Issue being fixed or feature implemented

Code updates made for ease of future `wasm-dpp` integration into JS services

## What was done?

- added missing `addOperation` method for `ExecutionContext`
- updated `DashPlatformProtocol` constructor in `wasm-dpp` to accept external entropy generator
- added `verifySignature` method to bindings of Identity Signed state transitions
- Fixed bug with `identity` instance move in `IdentityFacade#validate`
- Reworked wasmDpp JS loader to support parallel calls

## How Has This Been Tested?

- tested during the development of the following PR #848 

## Breaking Changes

None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/wasm-dpp/src/data_contract/state_transition/data_contract_create_transition/mod.rs [hunk 1]
```diff
@@ -169,4 +172,33 @@ impl DataContractCreateTransitionWasm {
...
+            .verify_signature(&identity_public_key.to_owned().into(), &bls_adapter);
+
+        match verification_result {
+            Ok(()) => Ok(true),
+            Err(protocol_error) => match &protocol_error {
+                ProtocolError::AbstractConsensusError(err) => match err.as_ref() {
+                    ConsensusSignatureErrorVariant(err) => match err {
+                        SignatureError::InvalidStateTransitionSignatureError => Ok(false),
+                        _ => Err(protocol_error),
+                    },
```
Comment (antouhou) [inline packages/wasm-dpp/src/data_contract/state_transition/data_contract_create_transition/mod.rs:196]: Looks like a linter warning :)

File: packages/wasm-dpp/src/data_contract/state_transition/data_contract_create_transition/mod.rs [hunk 2]
```diff
@@ -169,4 +172,33 @@ impl DataContractCreateTransitionWasm {
             )
             .with_js_error()
     }
+
+    // TODO: add verify_signature to other identity signed state transitions
+    #[wasm_bindgen(js_name=verifySignature)]
+    pub fn verify_signature(
```
Comment (antouhou) [inline packages/wasm-dpp/src/data_contract/state_transition/data_contract_create_transition/mod.rs:178]: I'm wondering where is it called, as it's also called inside the `::validate` function

File: packages/wasm-dpp/src/data_contract/state_transition/data_contract_update_transition/mod.rs [hunk 3]
```diff
@@ -159,4 +166,32 @@ impl DataContractUpdateTransitionWasm {
...
+            .verify_signature(&identity_public_key.to_owned().into(), &bls_adapter);
+
+        match verification_result {
+            Ok(()) => Ok(true),
+            Err(protocol_error) => match &protocol_error {
+                ProtocolError::AbstractConsensusError(err) => match err.as_ref() {
+                    ConsensusSignatureErrorVariant(err) => match err {
+                        SignatureError::InvalidStateTransitionSignatureError => Ok(false),
+                        _ => Err(protocol_error),
+                    },
```
Comment (antouhou) [inline packages/wasm-dpp/src/data_contract/state_transition/data_contract_update_transition/mod.rs:189]: Clippy warning

File: packages/wasm-dpp/src/document/state_transition/document_batch_transition/mod.rs [hunk 4]
```diff
@@ -296,12 +298,29 @@ impl DocumentsBatchTransitionWasm {
...
+            .verify_signature(&identity_public_key.to_owned().into(), &bls_adapter);
+
+        match verification_result {
+            Ok(()) => Ok(true),
+            Err(protocol_error) => match &protocol_error {
+                ProtocolError::AbstractConsensusError(err) => match err.as_ref() {
+                    ConsensusSignatureErrorVariant(err) => match err {
+                        SignatureError::InvalidStateTransitionSignatureError => Ok(false),
+                        _ => Err(protocol_error),
+                    },
```
Comment (antouhou) [inline packages/wasm-dpp/src/document/state_transition/document_batch_transition/mod.rs:317]: Clippy

File: packages/wasm-dpp/src/identity/state_transition/identity_update_transition/identity_update_transition.rs [hunk 5]
```diff
@@ -483,4 +487,32 @@ impl IdentityUpdateTransitionWasm {
...
+            .verify_signature(&identity_public_key.to_owned().into(), &bls_adapter);
+
+        match verification_result {
+            Ok(()) => Ok(true),
+            Err(protocol_error) => match &protocol_error {
+                ProtocolError::AbstractConsensusError(err) => match err.as_ref() {
+                    ConsensusSignatureErrorVariant(err) => match err {
+                        SignatureError::InvalidStateTransitionSignatureError => Ok(false),
+                        _ => Err(protocol_error),
+                    },
```
Comment (antouhou) [inline packages/wasm-dpp/src/identity/state_transition/identity_update_transition/identity_update_transition.rs:510]: Clippy :)

File: packages/rs-dpp/src/identity/state_transition/identity_create_transition/identity_create_transition.rs [hunk 6]
```diff
@@ -125,6 +125,12 @@ impl IdentityCreateTransition {
             state_transition.set_asset_lock_proof(AssetLockProof::try_from(proof)?)?;
         }
 
+        if let Some(signature) =
```
Comment (antouhou) [inline packages/rs-dpp/src/identity/state_transition/identity_create_transition/identity_create_transition.rs:128]: Was it missing from the port?

Comment (markin-io) [reply packages/rs-dpp/src/identity/state_transition/identity_create_transition/identity_create_transition.rs:128]: Yes

File: packages/wasm-dpp/src/document/state_transition/document_batch_transition/mod.rs (hunk 4 above)
Comment (markin-io) [reply packages/wasm-dpp/src/document/state_transition/document_batch_transition/mod.rs]: Fixed

File: packages/wasm-dpp/src/data_contract/state_transition/data_contract_update_transition/mod.rs (hunk 3 above)
Comment (markin-io) [reply packages/wasm-dpp/src/data_contract/state_transition/data_contract_update_transition/mod.rs]: Fixed

File: packages/wasm-dpp/src/data_contract/state_transition/data_contract_create_transition/mod.rs (hunk 1 above)
Comment (markin-io) [reply packages/wasm-dpp/src/data_contract/state_transition/data_contract_create_transition/mod.rs]: Fixed

**Review by shumkov — APPROVED** [review]
👍

**Pull request merged**